src/interface/blas2_interface.hpp

```diff
@@ -487,6 +487,43 @@ typename sb_handle_t::event_t _gbmv_impl(sb_handle_t& sb_handle, char _trans,
   }
 }
 
+template <uint32_t local_range, uplo_type uplo, transpose_type trn,
+          diag_type diag, typename sb_handle_t, typename index_t,
+          typename container_t0, typename container_t1, typename increment_t>
+typename sb_handle_t::event_t _tbmv_impl(sb_handle_t& sb_handle, index_t _N,
+                                         index_t _K, container_t0 _mA,
+                                         index_t _lda, container_t1 _vx,
+                                         increment_t _incx) {
+  constexpr bool is_upper = (uplo == uplo_type::Upper);
+  constexpr bool is_transposed = (trn != transpose_type::Normal);
+  constexpr bool is_unit = (diag == diag_type::Unit);
+
+  if (_K >= _N) {
+    throw std::invalid_argument("Erroneous parameter");
+  }
+
+  constexpr index_t one = 1;
+  auto x_vector_size = _N;
+  auto res_buffer =
+      blas::make_sycl_iterator_buffer<typename container_t0::scalar_t>(
+          x_vector_size);
+
+  auto mA = make_matrix_view<col_major>(_mA, _K + 1, _N, _lda);
+  auto vx = make_vector_view(_vx, _incx, x_vector_size);
+  auto vres = make_vector_view(res_buffer, one, x_vector_size);
+
+  const index_t global_size = roundUp<index_t>(x_vector_size, local_range);
+  auto tbmv = make_tbmv<local_range, is_upper, is_transposed, is_unit>(vres, mA,
+                                                                       _K, vx);
+
+  auto tbmvEvent =
+      sb_handle.execute(tbmv, static_cast<index_t>(local_range), global_size);
+
+  auto assignOp = make_op<Assign>(vx, vres);
```

Comment:
Hi @mehdi-goli, thanks for your review!

I took this approach in order to save space/unnecessary copies in case of `vx` having `incx>1` which is the most generic choice considering that this is a baseline implementation. Maybe, we can apply your suggestion for the architectures that have a fast built-in copy intrinsic.

Reply:
Yes that is fine, there might be cases specially when there is stride, not having coalesce write could affect the kernel, but we can address that later not in this pr